x/stakeibc/keeper/msg_server_rebalance_validators_test.go

```diff
@@ -23,6 +24,7 @@ func (s *KeeperTestSuite) SetupRebalanceValidators() RebalanceValidatorsTestCase
 	// Setup IBC
 	delegationIcaOwner := "GAIA.DELEGATION"
 	delegationChannelId := s.CreateICAChannel(delegationIcaOwner)
+	s.MsgServer = stakeibckeeper.NewMsgServerImpl(s.App.StakeibcKeeper)
```

Comment:
is this line necessary?

Reply:
Yes, it does actually. because the Setup() run in SetupTest() only setup for tests that need to use Stride only. If we move to tests that use ICA or IBC (need interact between 2 chains) then we'll have to have this line, otherwise the context that get pass from function to function will be different, either we have this line or we'll have to change code in CreateICAChannel which I choose to go with the 1st approach.

Reply:
Ah shoot. That's sounding familiar - that might be why we originally set it up as a function GetMsgServer (so that it gets executed after the setup)

I think its a confusing UX for someone to need to know that they have to reinitialize MsgServer every time they setup ICA/IBC tests. 

Can you think of a clean way we could update it in CreateICAChannel (or preferably in CreateTransferChannel, similar to how we update the Ctx)? I'm not sure there's an easy way since the msg server is at the module keeper level and app tester is at the app level. 

If there's no clean way, I'd unfortunately probably vote to leave it as a function GetMsgServer

Reply:
Eventhough I can manage to add an extra field to AppTestSuite with new param to get the module's type of keeper to make it work. I think it doesn't make sense to do it, I also checked other modules to see if the GetMsgServer get remove but none of it does. So maybe we should close this as complete and keep GetMsgServer. @sampocs

Reply:
Yeah I agree with all that. Sorry about sending you down this path :( 

Do you mind just adding a comment to each `GetMsgServer` function that describes this so we don't accidentally go down this road again in the future? You can do it in a separate PR

Something like
```
// Dynamically gets the MsgServer for this module's keeper
// A function must be used so that the MsgServer is always created with the most updated App context
//  which can change depending on the type of test 
//  (e.g. tests with only one Stride chain vs tests with multiple chains and IBC support) 
```

Reply:
Got it! I'll add comment on that

Reply:
Thank you!